pubspec.yaml

```diff
@@ -1,12 +1,15 @@
 name: math_expressions
-version: 2.3.0
+version: 2.3.1
 homepage: https://github.com/fkleon/math-expressions
 description: A library for parsing and evaluating mathematical expressions, supporting real numbers, vectors, and basic interval arithmetic.
 
 dependencies:
+  source_span: ^1.9.0
```

Comment:
I appreciate your parser is using `source_span` to generate nice error messages, but it does pull in quite a few transitive dependencies. Maybe it would be possible to add a parameter that allows to opt for simpler error message code without external dependencies?

Similar for the `tuple` dependency. A `Tuple2` which only acts as simple data holder could be implemented within the library or generated code.

I bring this up because I've been trying to keep this as library lean as possible in terms of dependencies.

Reply:
This is already possible.
I also plan to replace with a handmade version.

```dart
await fastBuild(
      context, [_parse, _refExpression], 'lib/src/experimental_parser.dart',
      header: __header, footer: __footer);
```

`addErrorMessage` in `addErrorMessageProcessor`

```dart
Future<void> fastBuild(Context context, List<Named> parsers, String filename,
    {bool addErrorMessageProcessor = true,
    String? footer,
    bool format = true,
    String? header,
    String? partOf,
    Map<String, Named> publish = const {}}) async {
```

```dart
/// An unofficial way to display error messages.
  /// Can be used as a starting point when developing parsers.
  ///
  /// An unofficial way because the concept of building a parser implies that
  /// the generated parser should not have dependencies.
  ///
  /// Use of this function requires the import of a third party package:
  ///
  /// import 'package:source_span/source_span.dart';
  static String getErrorMessageProcessor([String name = '_errorMessage']) {
    var result = _functionErrorMessage;
    result = result.replaceAll('{{name}}', name);
    return result;
  }
}
```

But I am at work now...

Reply:
>> Similar for the tuple dependency. A Tuple2 which only acts as simple data holder could be implemented within the library or generated code.

OK. Good idea.
But it will take a little time.
Especially since I'm not happy about it.
I'm waiting for support for Record types.